kernel/kernel-api/src/main/java/io/delta/kernel/internal/columndefaults/ColumnDefaults.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.columndefaults;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.actions.Metadata;
+import io.delta.kernel.internal.tablefeatures.TableFeatureSupport;
+import io.delta.kernel.internal.tablefeatures.TableFeatures;
+import io.delta.kernel.internal.util.SchemaIterable;
+import io.delta.kernel.types.*;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Utilities class for TableFeature "allowColumnDefaults". NOTE: As of Aug 2025, kernel only
+ * supports reading Delta tables with the table feature, or make metadata change to the table.
+ * Writing actual data to the table, or modify the default values is not allowed.
+ */
+public class ColumnDefaults {
+
+  private static final String DEFAULT_VALUE_METADATA_KEY = "CURRENT_DEFAULT";
+
+  /** Don't allow data writes to tables with the feature enabled */
+  public static void blockIfEnabled(Row transactionState) {
+    if (TableFeatureSupport.supports(
+        transactionState, TableFeatures.ALLOW_COLUMN_DEFAULTS_W_FEATURE)) {
+      throw new UnsupportedOperationException(
+          "Writing with Column Default values is not supported yet.");
+    }
+  }
+
+  /**
+   * Validate Column Default value changes in the provided metadata. Kernel does not support
+   * expressions for now and will treat all default values as literal.
+   */
+  public static void validate(Metadata newMetadata) {
+    for (SchemaIterable.SchemaElement element : new SchemaIterable(newMetadata.getSchema())) {
+      StructField field = element.getField();
+      if (field.getMetadata().contains(DEFAULT_VALUE_METADATA_KEY)) {
+        validateLiteral(
+            field.getDataType(), field.getMetadata().getString(DEFAULT_VALUE_METADATA_KEY));
```

Comment:
I think we want to validate only if the expression has changed right? In the current mode we always evaluate.